### PR TITLE
Draft: detect/pcre: Remove shared pcre2 match objects

### PR DESCRIPTION
**Pull request by jlucovsky**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [ ] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [ ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
-
-
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 15200

**Comment by victorjulien** [comment]
```
detect-config.c:307:9: warning: Branch condition evaluates to a garbage value [core.uninitialized.Branch]
672
    if (match) {
673
        ^~~~~
```

**Comment by victorjulien** [comment]
```
detect-filestore.c:486:9: warning: Branch condition evaluates to a garbage value [core.uninitialized.Branch]
788
    if (match) {
789
        ^~~~~
790
2 warnings generated.
```

File: src/detect-byte-extract.c [hunk 1]
```diff
@@ -507,10 +504,16 @@ static inline DetectByteExtractData *DetectByteExtractParse(DetectEngineCtx *de_
             bed->endian = DETECT_BYTE_EXTRACT_ENDIAN_DEFAULT;
     }
 
+    pcre2_match_data_free(match);
+    match = NULL;
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply src/detect-byte-extract.c:508]: fixed

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .tcp.overlap | 1124925 | 1221170 | 108.56% | 

Pipeline 15201

File: src/detect-flowint.c [hunk 2]
```diff
@@ -230,27 +230,28 @@ static DetectFlowintData *DetectFlowintParse(DetectEngineCtx *de_ctx, const char
...
     uint8_t modifier = FLOWINT_MODIFIER_UNKNOWN;
     unsigned long long value_long = 0;
     const char *str_ptr;
+    pcre2_match_data *match = NULL;
 
-    ret = DetectParsePcreExec(&parse_regex, rawstr, 0, 0);
+    int ret = DetectParsePcreExec(&parse_regex, &match, rawstr, 0, 0);
     if (ret < 3 || ret > 4) {
         SCLogError("\"%s\" is not a valid setting for flowint(ret = %d).", rawstr, ret);
         return NULL;
```
Comment (victorjulien) [inline src/detect-flowint.c:243]: leaks match if ret >= 0 but not 3 or 4, fixing

Comment (jlucovsky) [reply src/detect-flowint.c:243]: Good catch; fixing now.

**Comment by victorjulien** [comment]
replaced by #9242

**Pull request closed without merging**